### DISCUSSION TITLE
As many Etags as routes

### DISCUSSION THREAD

**Discussion by kael** [remix-run/remix#6555] [Q&A]
I'm trying to use as many Etags as there are routes.

I'm fetching two sets of data, one to be displayed as a `<main/>`, the other as an `<aside/>`, using `<Suspense/>` and `defer`.

The only way I have found to use multiple Etags is to create a composite Etag in the form of `W/"main:etag;aside:etag"` that I'm parsing and splitting upon requests, then matching the different states of caches for each Etag.

With the current architecture, it doesn't seem possible to use multiple Etags, but I'm wondering if using an [Islands Architecture #3861](https://github.com/remix-run/remix/discussions/3861) would allow such a case.

There's also a case of using [multiple Etags](https://stackoverflow.com/questions/41657773/why-do-the-http-precondition-header-fields-support-multiple-etags) in the form of `If-Match: "xyzzy", "r2d2xxxx", "c3piozzzz"` but Chrome seems to ignore Etags others than the first one (and I'm not sure of this behaviour anyway).

Any ideas how to use several Etags for multiple routes ?

**Comment by kiliman** [comment]
With Remix, you have the initial document request (which can comprise data from 1 or more loaders). That `Etag` should be a composite of all the loader `Etags`.

Subsequent navigations and revalidations will call individual loaders, which will return their own `Etags`.

The browser will add the `If-Match` headers depending on the request (full page vs individual loaders). So the full page request will have the composite `Etag` that you'll need to parse in your loaders.

**Reply by kael** [reply]
> Subsequent navigations and revalidations will call individual loaders, which will return their own `Etags`.

Thank you very much for the explanation, you cleared some misconception I had of how loaders work.

I had seen how the initial HTTP request is handled with the `Transfer-Encoding: chunked` header field (by using _Wireshark_), and how chunks of data are sent to the browser, but wasn't sure about subsequent navigations.

I'm gonna give a try, so.

Cheers.
